### ISSUE TITLE
[feature/changelog] Order Change Types Based on Bump Map Rules

### ISSUE THREAD

**Issue by KyleKing** [commitizen-tools/commitizen#319]
## Description

The `CHANGELOG` is ordered with the most recent change type first. This makes the diffs hard to read and may put less important change types over more important ones

Example diff from one of my projects: https://github.com/KyleKing/calcipy/commit/a61d0a4447f5ec03a3eccf279533be7955801256#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed

## Possible Solution

Could the `CHANGELOG` be ordered based on the bump rules?

The `CHANGELOG` could be ordered in descending importance from `(MAJOR, MINOR, PATCH, OTHER)` where the types within each section would be sorted alphabetically and the `OTHER` section would capture any change types shown in the `CHANGELOG`, but not part of the bump rules

## Related Issue

I looked through the issues (matching [order](https://github.com/commitizen-tools/commitizen/issues?q=order) or [changelog](https://github.com/commitizen-tools/commitizen/issues?q=changelog)) but didn't find any that appear to be asking for the same feature

## PR Notes

I would be happy to try to add this functionality, if this feature would be of interest to the maintainers. If not, I may look into separate [customization](https://commitizen-tools.github.io/commitizen/customization/#custom-changelog-generator) as an alternative to standard functionality) 

I created a [cz_legacy customization](https://github.com/KyleKing/cz_legacy) last night, so I have some familiarity with the commitizen code base

**Comment by Woile** [comment]
Hello Kyle! Thanks, that would be an awesome feature, feel free to create a PR with it. Also, you can add your custom rule to our docs in here: https://commitizen-tools.github.io/commitizen/third-party-commitizen/

**Comment by KyleKing** [comment]
I found a few small questions/issues that seemed too trivial for a whole issue:

- For the issues template, `python -c "import platform; print(platform.platform())"` might provide more useful debugging information (i.e. `macOS-10.15.7-x86_64-i386-64bit` vs. `Darwin`)
- In my fork, the format script made changes to import order in a number of files. It may be helpful to set `commitizen` as a known first party for isort
- (WIP) I ran into issues with Unicode characters in commit messages on Windows. I need to revisit and fully understand why `open` was defaulting to `cp1252` instead of Unicode. Initially, explicitly specifying the encoding fixed the issue, but I'm not sure if that is the best fix (`with open(self.file_name, "w") as changelog_file:`)
